lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -1829,6 +1840,32 @@ SPIRVValue *LLVMToSPIRV::transIndirectCallInst(CallInst *CI,
       BB);
 }
 
+SPIRVValue *LLVMToSPIRV::transAsmINTEL(InlineAsm *IA) {
+  assert(IA);
+
+  // TODO: intention here is to provide information about actual target
+  //       but in fact spir-64 is substituted as triple when translator works
+  //       eventually we need to fix it (not urgent)
+  StringRef TripleStr(M->getTargetTriple());
+  auto AsmTarget = static_cast<SPIRVAsmTargetINTEL *>(
+      BM->addAsmTargetINTEL(TripleStr.str()));
```

Comment:
Here you add `AsmTargetINTEL` with the same target for every `AsmINTEL`.  We should not have multiple instructions with the same operands. Like:
```


8 AsmTargetINTEL 31 "spir64-unknown-unknown"
12 AsmINTEL 7 33 32 31 "intcommand $0 $1""=r,r"

8 AsmTargetINTEL 58 "spir64-unknown-unknown"
12 AsmINTEL 35 60 59 58 "floatcommand $0 $1""=r,r"

8 AsmTargetINTEL 104 "spir64-unknown-unknown"
17 AsmINTEL 2 106 105 104 "mixed_integral_command $0 $3 $1 $2""=r,r,r,r"

8 AsmTargetINTEL 141 "spir64-unknown-unknown"
15 AsmINTEL 108 143 142 141 "mixed_floating_command $0 $1 $2""=r,r,r"

8 AsmTargetINTEL 179 "spir64-unknown-unknown"
16 AsmINTEL 62 181 180 179 "mixed_all_command $0 $3 $1 $2""=r,r,r,r"

8 AsmTargetINTEL 211 "spir64-unknown-unknown"
18 AsmINTEL 212 214 213 211 "multiple_command $0 $0 $1 $1 $2 $2""=r,=r,=r,0,1,2"

8 AsmTargetINTEL 228 "spir64-unknown-unknown"
11 AsmINTEL 6 230 229 228 "constcommand $0 $1""i,i"

```
 Instead all `AsmINTEL` should refer to one `AsmTargetINTEL`:
```
8 AsmTargetINTEL 31 "spir64-unknown-unknown"
12 AsmINTEL 7 33 32 31 "intcommand $0 $1""=r,r"
12 AsmINTEL 35 60 59 31 "floatcommand $0 $1""=r,r"
17 AsmINTEL 2 106 105 31 "mixed_integral_command $0 $3 $1 $2""=r,r,r,r"
15 AsmINTEL 108 143 142 31 "mixed_floating_command $0 $1 $2""=r,r,r"
16 AsmINTEL 62 181 180 31 "mixed_all_command $0 $3 $1 $2""=r,r,r,r"
18 AsmINTEL 212 214 213 31 "multiple_command $0 $0 $1 $1 $2 $2""=r,=r,=r,0,1,2"
11 AsmINTEL 6 230 229 31 "constcommand $0 $1""i,i"
```